infra.bs

```diff
@@ -186,6 +186,79 @@ caller's algorithm. Using prose the caller has the ability to "catch" the except
 another action.
 
 
+<h3 id=algorithm-conditional-abort>Conditional abort</h3>
+
+<p>Sometimes it is useful to stop performing a series of steps once some condition becomes true.
+
+<p>To do this, state that a given series of steps will <dfn export>abort when</dfn> a
+specific <var>condition</var> is reached. This indicates that the specified steps should be
+evaluated, not as-written, but by additionally inserting a step before each of them and after each
+of them except the last, that evaluates <var>condition</var>, and if <var>condition</var> evaluates
```

Comment:
If we just said "before each of them" I think it ends up being the same and clearer?

Reply:
Although I wonder if that's good enough. If the last step is computationally expensive it would be okay to abort during that step too...

Reply:
Good point about before.

I think it's OK; if there's an abort before the step, and there are no observable side effects of the step, you can abort during the step. Allowing you to abort after the step doesn't give you any extra freedom, I don't think.